src/flows/components/header/PublishedVersions.tsx

```diff
@@ -0,0 +1,47 @@
+import React, {FC, useCallback, useContext, useEffect, useState} from 'react'
+import VersionBullet from 'src/flows/components/header/VersionBullet'
+import {FlowContext} from 'src/flows/context/flow.current'
+import {isFlagEnabled} from 'src/shared/utils/featureFlag'
+import {
+  getNotebooksVersions,
+  VersionHistories,
+} from 'src/client/notebooksRoutes'
+
+const PublishedVersions: FC = () => {
+  const {flow} = useContext(FlowContext)
+  const [versions, setVersions] = useState<VersionHistories>([])
+
+  const handleGetNotebookVersions = useCallback(async () => {
+    try {
+      const response = await getNotebooksVersions({id: flow.id})
+
+      if (response.status !== 200) {
+        throw new Error(response.data.message)
+      }
+
+      setVersions(response.data)
+    } catch (error) {
+      console.error({error})
+    }
+  }, [flow])
+
+  useEffect(() => {
+    if (isFlagEnabled('flowPublishLifecycle')) {
+      handleGetNotebookVersions()
+    }
+  }, [handleGetNotebookVersions])
```

Comment:
this doesn't refetch after a notebook has been published. Besides prop drilling or an emitting an event. This needs to be worked on in order to trigger a call of the version history so that we can refetch the notebook published history when a user publishes a new version. This might just be using context, but it might be something else

Reply:
Moved everything to context so we can refetch there